vendor/luajwtjitsi/luajwtjitsi.lua

```diff
@@ -26,82 +26,83 @@ end
+    ['HS512'] = function(data, key) return hmac.new(key, 'sha512'):final(data) end,
+    ['RS256'] = function(data, key) return signRS(data, key, 'sha256') end,
+    ['RS384'] = function(data, key) return signRS(data, key, 'sha384') end,
+    ['RS512'] = function(data, key) return signRS(data, key, 'sha512') end
 }
 
 local alg_verify = {
-	['HS256'] = function(data, signature, key) return signature == alg_sign['HS256'](data, key) end,
-	['HS384'] = function(data, signature, key) return signature == alg_sign['HS384'](data, key) end,
-	['HS512'] = function(data, signature, key) return signature == alg_sign['HS512'](data, key) end,
-	['RS256'] = function(data, signature, key) return verifyRS(data, signature, key, 'sha256') end,
-	['RS384'] = function(data, signature, key) return verifyRS(data, signature, key, 'sha384') end,
-	['RS512'] = function(data, signature, key) return verifyRS(data, signature, key, 'sha512') end
+    ['HS256'] = function(data, signature, key) return signature == alg_sign['HS256'](data, key) end,
+    ['HS384'] = function(data, signature, key) return signature == alg_sign['HS384'](data, key) end,
+    ['HS512'] = function(data, signature, key) return signature == alg_sign['HS512'](data, key) end,
+    ['RS256'] = function(data, signature, key) return verifyRS(data, signature, key, 'sha256') end,
+    ['RS384'] = function(data, signature, key) return verifyRS(data, signature, key, 'sha384') end,
+    ['RS512'] = function(data, signature, key) return verifyRS(data, signature, key, 'sha512') end
 }
 
 -- Splits a token into segments, separated by '.'.
 -- @param token The full token to be split.
 -- @return A table of segments.
 local function split_token(token)
-	local segments={}
-  for str in string.gmatch(token, "([^\\.]+)") do
-    table.insert(segments, str)
-  end
-  return segments
+    local segments={}
+    for str in string.gmatch(token, "([^\\.]+)") do
+        table.insert(segments, str)
+    end
+    return segments
 end
 
 -- Parses a JWT token into it's header, body, and signature.
 -- @param token The JWT token to be parsed.
 -- @return A JSON header and body represented as a table, and a signature.
 local function parse_token(token)
-	local segments=split_token(token)
-  if #segments ~= 3 then
-		return nil, nil, nil, "Invalid token"
-	end
-
-	local header, err = cjson_safe.decode(basexx.from_url64(segments[1]))
-	if err then
-		return nil, nil, nil, "Invalid header"
-	end
-
-	local body, err = cjson_safe.decode(basexx.from_url64(segments[2]))
-	if err then
-		return nil, nil, nil, "Invalid body"
-	end
-
-	local sig, err = basexx.from_url64(segments[3])
-	if err then
-		return nil, nil, nil, "Invalid signature"
-	end
-
-	return header, body, sig
+    local segments = split_token(token)
+    if #segments ~= 3 then
+        return nil, nil, nil, "Invalid token"
+    end
+    local err, header, body, sig
+
+    header, err = cjson_safe.decode(basexx.from_url64(segments[1]))
```

Comment:
This is gonna be confusing if we ever update this vendor lib because we're not the author of this ...

Reply:
Ah !